src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -0,0 +1,186 @@
+        }
+
+        /// <summary>
+        /// Checks if a subscription has registered the ChangeAnalysis RP.
+        /// </summary>
+        /// <param name="subscriptionId">Subscription Id.</param>
+        public async Task<SubscriptionOnboardingStatus> CheckSubscriptionOnboardingStatus(string subscriptionId)
+        {
+            string requestUri = changeAnalysisEndPoint + $"Subscription/{subscriptionId}";
+            try
+            {
+               string jsonString = await PrepareAndSendRequest(requestUri, httpMethod: HttpMethod.Get);
+               return JsonConvert.DeserializeObject<SubscriptionOnboardingStatus>(jsonString);
+            }
+            catch (Exception)
+            {
+                return new SubscriptionOnboardingStatus
+                {
+                    IsRegistered = false
+                };
+            }
+        }
+
+        /// <summary>
+        /// Prepares httpwebrequest to <paramref name="requestUri"/> with <paramref name="postBody"/> as body of the request.
+        /// </summary>
+        /// <param name="requestUri">Change Analysis Request URI</param>
+        /// <param name="postBody">Body of the request</param>
+        /// <returns>JSON string received from <paramref name="requestUri"/>.</returns>
+        private async Task<string> PrepareAndSendRequest(string requestUri, object postBody = null, HttpMethod httpMethod = null)
+        {
+            HttpMethod requestedHttpMethod = httpMethod == null ? HttpMethod.Post : httpMethod;
+            HttpRequestMessage requestMessage = new HttpRequestMessage(requestedHttpMethod, requestUri);
+            string authToken = await ChangeAnalysisTokenService.Instance.GetAuthorizationTokenAsync();
+
+            // Add required headers.
+            requestMessage.Headers.Add("Authorization", authToken);
+            requestMessage.Headers.Add("x-ms-client-object-id", clientObjectIdHeader);
+
+            // For requests coming from Diagnose and Solve, add x-ms-principal-name header.
+            if (!string.IsNullOrWhiteSpace(clientPrincipalNameHeader))
+            {
+               requestMessage.Headers.Add("x-ms-principal-name", clientPrincipalNameHeader);
+            }
+
+            if (postBody != null)
+            {
+               requestMessage.Content = new StringContent(JsonConvert.SerializeObject(postBody), Encoding.UTF8, "application/json");
+            }
+
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(DataProviderConstants.DefaultTimeoutInSeconds));
+            HttpResponseMessage responseMessage = await httpClient.SendAsync(requestMessage, cancellationTokenSource.Token);
+            string content = await responseMessage.Content.ReadAsStringAsync();
+            if (!responseMessage.IsSuccessStatusCode)
+            {
+                throw new Exception(responseMessage.ReasonPhrase);
```

Comment:
If you are throwing an exception here, consider following:
- Use an HttpException
- Add Details in the exception like Status Code (Reason Phrase) , Content so that its easy to debug